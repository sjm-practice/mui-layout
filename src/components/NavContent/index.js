import React from "react";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListSubheader from "@material-ui/core/ListSubheader";
import Divider from "@material-ui/core/Divider";
import DashboardIcon from "@material-ui/icons/Dashboard";
import ShoppingCartIcon from "@material-ui/icons/ShoppingCart";
import PeopleIcon from "@material-ui/icons/People";
import BarChartIcon from "@material-ui/icons/BarChart";
import LayersIcon from "@material-ui/icons/Layers";
import AssignmentIcon from "@material-ui/icons/Assignment";

const FunctionsList = () => (
  <List>
    <ListItem button>
      <ListItemIcon>
        <DashboardIcon />
      </ListItemIcon>
      <ListItemText primary="Dashboard" />
    </ListItem>
    <ListItem button>
      <ListItemIcon>
        <PeopleIcon />
      </ListItemIcon>
      <ListItemText primary="User Management" />
    </ListItem>
    <ListItem button>
      <ListItemIcon>
        <BarChartIcon />
      </ListItemIcon>
      <ListItemText primary="Reports" />
    </ListItem>
  </List>
);

const SavedReportsList = () => (
  <List>
    <ListSubheader inset>Saved reports</ListSubheader>
    <ListItem button>
      <ListItemIcon>
        <AssignmentIcon />
      </ListItemIcon>
      <ListItemText primary="Current month" />
    </ListItem>
    <ListItem button>
      <ListItemIcon>
        <AssignmentIcon />
      </ListItemIcon>
      <ListItemText primary="Last quarter" />
    </ListItem>
  </List>
);

const NavContent = () => (
  <>
    <FunctionsList />
    <Divider />
    <SavedReportsList />
  </>
);

export default NavContent;
