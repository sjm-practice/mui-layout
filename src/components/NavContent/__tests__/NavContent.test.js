/* eslint-env jest */

import React from "react";
import { render } from "@testing-library/react";

import NavContent from "../index";

describe("<NavContent />", () => {
  it("Should match Snapshot", () => {
    const { asFragment } = render(<NavContent />);

    expect(asFragment()).toMatchSnapshot();
  });
});
