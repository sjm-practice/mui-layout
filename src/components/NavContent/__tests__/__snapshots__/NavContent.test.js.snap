// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavContent /> Should match Snapshot 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root MuiList-padding"
  >
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button Mui-selected"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          folder
        </span>
      </div>
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap"
        >
          My Files
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          people
        </span>
      </div>
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap"
        >
          Shared with me
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          star
        </span>
      </div>
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap"
        >
          Starred
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          schedule
        </span>
      </div>
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap"
        >
          Recent
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          offline_pin
        </span>
      </div>
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap"
        >
          Offline
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          publish
        </span>
      </div>
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap"
        >
          Uploads
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          backup
        </span>
      </div>
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap"
        >
          Backups
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          delete
        </span>
      </div>
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap"
        >
          Trash
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <hr
      class="MuiDivider-root"
      style="margin: 12px 0px;"
    />
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          settings
        </span>
      </div>
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-noWrap"
        >
          Settings & account
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </ul>
</DocumentFragment>
`;
